skxray/core/correlation.py

```diff
@@ -253,6 +253,211 @@ def multi_tau_auto_corr(num_levels, num_bufs, labels, images):
+
+        # to track how many images processed in each level
+        img_per_level = np.zeros(num_levels, dtype=np.int64)
+
+        start_time = time.time()  # used to log the computation time (optionally)
+
+    else:
+        G = previous[2]
+        past_intensity_norm = previous[3]
+        future_intensity_norm = previous[4]
+        buf = previous[5]
+
+    for n, img in enumerate(images):
+
+        cur[0] = (1 + cur[0]) % num_bufs  # increment buffer
+
+        # Put the image into the ring buffer.
+        buf[0, cur[0] - 1] = (np.ravel(img))[pixel_list]
+
+        # Compute the correlations between the first level
+        # (undownsampled) frames. This modifies G,
+        # past_intensity_norm, future_intensity_norm,
+        # and img_per_level in place!
+        _process(buf, G, past_intensity_norm,
+                 future_intensity_norm, label_mask,
+                 num_bufs, num_pixels, img_per_level,
+                 level=0, buf_no=cur[0] - 1)
+
+        # check whether the number of levels is one, otherwise
+        # continue processing the next level
+        processing = num_levels > 1
+
+        # Compute the correlations for all higher levels.
+        level = 1
+        while processing:
+            if not track_level[level]:
+                track_level[level] = 1
+                processing = False
+            else:
+                prev = 1 + (cur[level - 1] - 2) % num_bufs
+                cur[level] = 1 + cur[level] % num_bufs
+
+                buf[level, cur[level] - 1] = (buf[level - 1, prev - 1] +
+                                              buf[level - 1,
+                                                  cur[level - 1] - 1])/2
+
+                # make the track_level zero once that level is processed
+                track_level[level] = 0
+
+                # call the _process function for each multi-tau level
+                # for multi-tau levels greater than one
+                # Again, this is modifying things in place. See comment
+                # on previous call above.
+                _process(buf, G, past_intensity_norm,
+                         future_intensity_norm, label_mask,
+                         num_bufs, num_pixels, img_per_level,
+                         level=level, buf_no=cur[level]-1,)
+                level += 1
+
+                # Checking whether there is next level for processing
+                processing = level < num_levels
+
+    # ending time for the process
+    end_time = time.time()
+
+    logger.info("Processing time for {0} images took {1} seconds."
+                "".format(n, (end_time - start_time)))
+
+    # the normalization factor
+    if len(np.where(past_intensity_norm == 0)[0]) != 0:
+        g_max = np.where(past_intensity_norm == 0)[0][0]
+    else:
+        g_max = past_intensity_norm.shape[0]
+
+    # g2 is normalized G
+    g2 = (G[:g_max] / (past_intensity_norm[:g_max] *
+                       future_intensity_norm[:g_max]))
+
```

Comment:
I think 'yield' can come here and the tot_channels and lag_steps are taken out from this function. Because we can find it separately